.github/CONTRIBUTING.md

```diff
@@ -181,20 +247,67 @@ are browsing the repository.
 
 If you're feeling adventurous, we're interested in adding a syntax section to each
 README. Every time a unique language syntax appears in the repo for a particular
-language, we track it in its README with links. Check out the [Python README][13]
+language, we track it in its README with links. Check out the [Python README][python-readme]
 for an example.
 
-Don't worry if you forget any of this; we have a checklist of reminders in the
-pull request template. At any rate, let's have some fun!
+#### Add `testinfo.yml`
+
+In addition to the README, all new languages must be setup for testing.
+To do so, add a file called `testinfo.yml` to the new language directory.
 
-### Tests
+Refer to the [Glotter Wiki][glotter-directory-config] and the [Tests in Detail section][tests-in-detail] below
+for more information about the contents of that file.
+
+## Issues in Detail
+
+Any issues for code or tests must reference the relevant project.
+To do so go to the [Projects List page][projects-list] on the Sample Programs website.
+Find the project page from the list of projects there and paste a link to that page into the issue.
+
+For simplicity, we ask that you **only make one issue for one language and one project at a time.**
+This and the project link help is tagging and maintaining issues and pull requests.
+(See [the pull request section above][add-one-and-only-one-snippet-at-a-time] for the pull request side of this flow)
+
+As such please name the issue in the format "{Add/modify} {project name} in {language}."
+For example if I were to add a new snippet for the fibonacci project in python,
+I would name my issue "Add Fibonacci in Python".
+If I were to modify an existing snippet for the fibonacci project in python,
+I would name my issue "Modify Fibonacci in Python".
+
+Issues that are too broad will be closed and or split into multiple issues that follow proper naming format.
+
+### Modifying Existing Code Snippets
+
+The intended purpose of this repository is to promote learning and sharing of programming languages.
+In order to do so, we have intentionally selected relatively small projects that can be implemented as simple single file code snippets.
+We realize that there are many different ways to implement these projects even within the same language.
+
+Some existing implementations may not be the "best" or "most efficient" implementation.
+We are okay with that as long as the code snippet still helps with our goal to promote learning.
+
+That means that **we will not accept issues or pull requests to modify existing snippets**
+unless you are able to explain in the issue why the existing implementation does not follow the guidelines specified
+in the [project documentation][project-list] or why the existing implementation detracts from the goal of this repository in some way.
+
+If you are making such a claim, we recommend waiting for approval from the core team (`@TheRenegadeCoder/core`) for beginning work on a pull request.
```

Comment:
This is really great! I should have been following something like this previously.